stablehlo/integrations/python/tests/stablehlo_to_tf_saved_model_test.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 The StableHLO Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mlir.dialects.stablehlo as stablehlo
+import mlir.ir as ir
+from mlir.stablehlo_to_tf_saved_model import InputLocation, save_stablehlo_as_tf_saved_model
+import numpy as np
+import tensorflow as tf
+
+
+# Convert a stablehlo program, expressing a nn.Lienar layer with constant values
+# for weight and bias, to saved model
+
+mlir_module_string = """
+  module @IrToHlo.12 attributes {mhlo.cross_program_prefetches = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false} {
+
+  func.func @main(%arg0: tensor<2xf32>, %arg1: tensor<2x2xf32>, %arg2: tensor<2x2xf32>) -> tensor<2x2xf32> {
+    %0 = stablehlo.transpose %arg1, dims = [1, 0] {result_layout = dense<[0, 1]> : tensor<2xindex>, xla_shape = "f32[2,2]{0,1}"} : (tensor<2x2xf32>) -> tensor<2x2xf32>
+    %1 = stablehlo.dot_general %arg2, %0, contracting_dims = [1] x [0], precision = [DEFAULT, DEFAULT] : (tensor<2x2xf32>, tensor<2x2xf32>) -> tensor<2x2xf32>
+    %2 = stablehlo.broadcast_in_dim %arg0, dims = [1] : (tensor<2xf32>) -> tensor<2x2xf32>
+    %3 = stablehlo.add %1, %2 : tensor<2x2xf32>
+    return %3 : tensor<2x2xf32>\n
+  }
+}
+"""
+
+ctx = ir.Context()
+stablehlo.register_dialect(ctx)
+module = ir.Module.parse(mlir_module_string, ctx)
+
+input_locations = [
+    InputLocation.parameter(name='linear_layer.bias'),
+    InputLocation.parameter(name='linear_layer.weight'),
+    InputLocation.input_arg(position=0),
```

Comment:
Do we need to specify input shape anywhere?

Reply:
The input shapes are derived from within the `save_stablehlo_as_tf_saved_model` using the module parameter of the API.